### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by hisapy**
Includes:

* Refined umbrella architecture
* Webpack code-splitting 
* Yarn
* Other improvements

### CODE REVIEW

File: ui/star_wars/src/AppLayout.js [hunk 1]
```diff
@@ -28,7 +28,11 @@ class AppLayout extends React.Component {
             <LayoutTitle>Star Wars example</LayoutTitle>
             <LayoutSpacer />
             <Navigation>
-              <NavigationLink to="/">Home</NavigationLink>
+              {/*
+                TODO: Use <NavigationLink tag='a' href='star-wars'> after
+                upgrading react-to-mdl.
+              */}
```
Comment (rsjavier14) [inline ui/star_wars/src/AppLayout.js]: The "tag" parameter will not be necessary with the new react-to-mdl update.

File: ui/webpack.config.js [hunk 2]
```diff
@@ -22,43 +37,23 @@ module.exports = {
...
-      loader: ExtractTextPlugin.extract(
-        'style?sourceMap',
-        [
-          'css?modules&importLoaders=1&localIdentName=[path]___[name]__[local]___[hash:base64:5]',
-          'resolve-url',
-          'sass?sourceMap'
-        ]
-      )
-    },{
-      test: /\.json$/,
```
Comment (rsjavier14) [inline ui/webpack.config.js]: This is not necessary for react-markdown?

File: ui/package.json [hunk 3]
```diff
@@ -28,13 +29,13 @@
...
     "exports-loader": "^0.6.3",
-    "extract-text-webpack-plugin": "^1.0.1",
+    "extract-text-webpack-plugin": "2.0.0-beta.4",
     "node-sass": "^4.0.0",
     "resolve-url-loader": "^1.6.1",
     "sass-loader": "^4.1.0",
     "style-loader": "^0.13.1",
     "sync-request": "^3.0.1",
-    "webpack": "^1.14.0"
+    "webpack": "2.1.0-beta.28"
```
Comment (rsjavier14) [inline ui/package.json:37]: Is it recommended to use beta version?

File: ui/core/src/AppLayout.js [hunk 4]
```diff
@@ -27,7 +27,11 @@ export default class extends React.Component {
             <LayoutTitle>pwr2-docker</LayoutTitle>
             <LayoutSpacer />
             <Navigation>
-              <NavigationLink to="/admin">Inicio</NavigationLink>
+              <NavigationLink to="/">Inicio</NavigationLink>
+              {/*
+                TODO: Use <NavigationLink tag='a' href='star-wars'> after
+                upgrading react-to-mdl.
+              */}
```
Comment (rsjavier14) [inline ui/core/src/AppLayout.js]: Same as above, tag is not necessary

File: ui/core/src/AppLayout.js [hunk 5]
```diff
@@ -30,7 +30,7 @@ export default class extends React.Component {
                 TODO: Use <NavigationLink tag='a' href='star-wars'> after
                 upgrading react-to-mdl.
               */}
```
Comment (rsjavier14) [inline ui/core/src/AppLayout.js]: "tag" is not necesary

File: ui/package.json (hunk 3 above)
Comment (hisapy) [reply ui/package.json:37]: Not actually recommended but it is working based on http://javascriptplayground.com/blog/2016/10/moving-to-webpack-2/

**Review by rsjavier14 — APPROVED** [review]

**Pull request merged**